### ISSUE TITLE
calwebb_tso3 error in extract_1d for NIRISS SOSS

### ISSUE THREAD

**Issue by stscijgbot-jp** [spacetelescope/jwst#4961]
_Issue [JP-1462](https://jira.stsci.edu/browse/JP-1462) was created on JIRA by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse):_

calwebb_tso3 processing of a NIRISS SOSS observation is failing in the extract_1d step with:
```java
2020-05-17 04:15:18,704 - stpipe.Tso3Pipeline.extract_1d - INFO - Beginning loop over 166 integrations ...
2020-05-17 04:15:18,825 - stpipe.Tso3Pipeline.extract_1d - WARNING - Inverse wcs is not implemented, so can't use target coordinates to get location of spectrum.
2020-05-17 04:16:43,874 - stpipe.Tso3Pipeline.extract_1d - INFO - ... 50 integrations done
2020-05-17 04:17:41,796 - stpipe.Tso3Pipeline.extract_1d - WARNING - /dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/astropy/modeling/fitting.py:492: RuntimeWarning: invalid value encountered in true_divide
  lhs /= scl


Traceback (most recent call last):
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/bin/strun", line 26, in <module>
    step = Step.from_cmdline(sys.argv[1:])
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/jwst/stpipe/step.py", line 156, in from_cmdline
    return cmdline.step_from_cmdline(args)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/jwst/stpipe/cmdline.py", line 346, in step_from_cmdline
    step.run(*positional)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/jwst/stpipe/step.py", line 392, in run
    step_result = self.process(*args)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/jwst/pipeline/calwebb_tso3.py", line 147, in process
    result = self.extract_1d(cube)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/jwst/stpipe/step.py", line 392, in run
    step_result = self.process(*args)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/jwst/extract_1d/extract_1d_step.py", line 216, in process
    was_source_model=False)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/jwst/extract_1d/extract.py", line 2584, in run_extract1d
    apply_nod_offset, was_source_model)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/jwst/extract_1d/extract.py", line 3068, in do_extract1d
    prev_offset, verbose, extract_params)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/jwst/extract_1d/extract.py", line 3585, in extract_one_slit
    extract_model.extract(data, wl_array, verbose)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/jwst/extract_1d/extract.py", line 1992, in extract
    weights=None)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/jwst/extract_1d/extract1d.py", line 214, in extract1d
    temp_image, x, j, bkglim, bkg_order
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/jwst/extract_1d/extract1d.py", line 447, in _fit_background_model
    y, val, weights=wht)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/astropy/modeling/fitting.py", line 175, in wrapper
    return func(self, model, x, y, z=z, **kwargs)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/astropy/modeling/fitting.py", line 506, in __call__
    rhs[good], rcond)
  File "<__array_function__ internals>", line 6, in lstsq
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/numpy/linalg/linalg.py", line 2259, in lstsq
    x, resids, rank, s = gufunc(a, b, rcond, signature=signature, extobj=extobj)
ValueError: On entry to DLASCL parameter number 4 had an illegal value
----------------------------------------------------------------------
ERROR RUNNING STEP 'Tso3Pipeline':
    On entry to DLASCL parameter number 4 had an illegal value
----------------------------------------------------------------------
```

**Comment by stscijgbot-jp** [comment]
_Comment by [James White](https://jira.stsci.edu/secure/ViewProfile.jspa?name=jwhite) on [JIRA](https://jira.stsci.edu/browse/JP-1462?focusedCommentId=465019#comment-465019):_

The issue seems to be coming from where extract1d attempts to calculate and extract background.

 

The background is calculated using the `_fit_background_model` function which produces a polynomial fit model for the background (which is used to approximate and subtract the background in the extraction region per column).

 

For the data referenced here, there is at least one case where for a particular column, the number of data points available for constructing the polynomial model (after removing "bad" values) is one, which results in an error when attempting to construct the model. 

 

To fix this, I've simply increased the necessary number of values to 2, which allows the extract_1d step to run to completion and skips background subtraction in later extraction.

**Comment by stscijgbot-jp** [comment]
_Comment by [James White](https://jira.stsci.edu/secure/ViewProfile.jspa?name=jwhite) on [JIRA](https://jira.stsci.edu/browse/JP-1462?focusedCommentId=465022#comment-465022):_

Background model: 

<https://github.com/spacetelescope/jwst/blob/839efc937484297b7c71f5bf1de851239d3dbdda/jwst/extract_1d/extract1d.py#L392>

 

Used in extract1d:

<https://github.com/spacetelescope/jwst/blob/839efc937484297b7c71f5bf1de851239d3dbdda/jwst/extract_1d/extract1d.py#L213>

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-1462?focusedCommentId=465826#comment-465826):_

Fixed by #4970
